poc/OwinTestApp/src/OwinClassLibrary/Utilities/MAuthClient.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Net.Http;
+using System.Threading.Tasks;
+using OwinClassLibrary.Authentication;
+using OwinClassLibrary.Cat;
+using OwinClassLibrary.Zipkin;
+
+namespace OwinClassLibrary.Utilities
+{
+    public class MAuthClient : IMAuthClient
+    {
+        private readonly HttpClient _client;
+
+        public MAuthClient(ITraceProvider provider)
+        {
+            _client = new HttpClient(new MAuthMessageHandler(new ZipkinMessageHandler(provider)));
+        }
+
+        public MAuthClient(ITraceProvider provider, HttpMessageHandler handler)
+        {
+            _client = new HttpClient(new MAuthMessageHandler(new ZipkinMessageHandler(provider, handler)));
+        }
+
+        public async Task<string> MakeRequest(string uri)
+        {
+            var response = await _client.GetAsync(uri).ConfigureAwait(false);
```

Comment:
use ConfigureAwait(false) to avoid deadlock if you use `MakeRequest().Result` in Action method. (Rave case)